### PR TITLE
fix(esl-event-listener): change resolving strategy and subscription process

### PR DESCRIPTION
**Pull request by ala-n**
**NOTE:**
Tests and docs under `src/modules/esl-event-listener` are not ready yet (just a part of the tests is effective, docs should be extended with a detailed description).
The implementation is ready to be reviewed.

### CODE REVIEW

File: src/modules/esl-utils/dom/events/misc.ts [hunk 1]
```diff
@@ -42,3 +42,17 @@ export const getOffsetPoint = (el: Element): Point => {
   const x = props.left + window.scrollX;
   return {x, y};
 };
+
+/**
+ * Splits and deduplicate event string
```
Comment (NastaLeo) [inline src/modules/esl-utils/dom/events/misc.ts:47]:
```suggestion
 * Splits and deduplicates event string
```

File: src/modules/esl-event-listener/core/descriptor.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import type {PropertyProvider} from '../../esl-utils/misc/functions';
+
+/** Sting CSS selector to find the target or {@link EventTarget} object or array of {@link EventTarget}s */
```
Comment (NastaLeo) [inline src/modules/esl-event-listener/core/descriptor.ts:3]:
```suggestion
/** String CSS selector to find the target or {@link EventTarget} object or array of {@link EventTarget}s */
```

File: src/modules/esl-event-listener/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+EventUtils.dispatch(el: EventTarget, eventName: string, eventInit?: CustomEventInit)
+```
+
+**Parameters**:
+
+`el` - Target element;
+
+`eventName` - Name of the event;
+
+`eventInit` - An object that specifies characteristics about the event. This parameter can be used to everwrite the default behavior of bubbling and being cancelable.
```
Comment (NastaLeo) [inline src/modules/esl-event-listener/README.md:24]:
```suggestion
`eventInit` - An object that specifies characteristics about the event. This parameter can be used to overwrite the default behavior of bubbling and being cancelable.
```

File: src/modules/esl-event-listener/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+```typescript
+EventUtils.subscribe(target: HTMLElement, eventDesc?: string | Partial<ESLListenerDescriptor>, handler: ESLListenerHandler): ESLEventListener[]
+```
+
+**Parameters**:
+
+`target` - An element to subsribe;
+
+`eventDesc` - Event type or obejct of evnt description data;
```
Comment (NastaLeo) [inline src/modules/esl-event-listener/README.md:63]:
```suggestion
`eventDesc` - Event type or object of event description data;
```

**Comment by ala-n** [comment]
The final docs will be proposed separately. Currently super short, and the early draft is here.
Tests are not ready yet as well, there are almost no dummy tests  + a lot of TODOs regarding tests.
Proposed to be merged to beta to unlock lust technical updates with the final cleanups regarding tests and docs.

**Review by NastaLeo — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Pull request merged**